### PR TITLE
feat: add pipeline list cmd

### PR DESCRIPTION
**Pull request by andreafalzetti**
# Proposed changes

Fixes https://github.com/okteto/app/issues/6999

Adds a command `okteto pipeline list` to list all dev environments.

### 1 - Happy Path

1. Build this branch
2. Using an Okteto Cluster, create a new namespace:
```
okteto ns <xyz>
```
3. Deploy the following pipelines:
```bash
okd pipeline deploy --repository https://github.com/okteto/movies --label=movies --label=demo
okd pipeline deploy --repository https://github.com/okteto/go-getting-started --label=go --label=demo
```
4. List pipelines:
```
okteto pipeline list
```
5. Validate that you have both dev environments with the right labels
6. Filter by one of the labels and observe if the table gets updated accordingly.
7. Now destroy one pipeline and quickly run the list command again, to make sure the Status is showing correctly
8. Then, using the UI put to sleep the namespace and run the list pipeline command again, notice how the status is now set to "Sleeping"

### 2 - Flag: Output
1. If you followed the steps above, continue listing in JSON with:
```
okteto pipeline list --output json
```
2. Now YAML
```
okteto pipeline list --output yaml
```
3. Try filtering by label to make sure the list is returned correctly filtered
4. Consider piping the output of the command into `jq` and `yq` to validate it's only outputting JSON or YAML without any other unwanted char.
5. Create a new namespace 
```
okteto ns <xyz>
```
6. List again and validate both JSON and YAML outputs return empty lists
```
okteto pipeline list --namespace <new>
```

### 3 - Flag: Context

1. At this point you should have a context that has at least 2 pipelines deployed. Run the list for another cluster by doing:
```
okteto pipeline list --context <ANOTHER OKTETO CLUSTER>
```
2. Observe that the list returned is correct (use the UI of the other cluster to validate)
3. Validate that your default context has not been changed, the flag should not have that side effect

### 4 - Flag: Namespace

1. Create a new namespace and switch to it using `okteto ctx <new ns>`
2. Now list pipelines of another namespace where you have something already deployed: `okteto pipeline list --namespace <name>`
3. Now try listing a namespace of another cluster:
```
okteto pipeline list --cluster <c> --namespace <n>
```
4. Make sure the data returned is correct, your default context and namespace have not been modified

### 5 - Vanilla Clusters

1. Try running the command against a vanilla cluster. You should see an error, not allowing you.

## CLI Quality Reminders 🔧

For both authors and reviewers:

- Scrutinize for potential regressions
- Ensure key automated tests are in place
- Build the CLI and test using the validation steps
- Assess Developer Experience impact (log messages, performances, etc)
- If too broad, consider breaking into smaller PRs
- Adhere to our [code style](https://github.com/okteto/okteto/blob/master/docs/code-style.md) and [code review](https://github.com/okteto/okteto/blob/master/docs/code-review.md) guidelines

### CODE REVIEW

File: cmd/pipeline/list.go [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+			c, _, err := pc.k8sClientProvider.Provide(okteto.Context().Cfg)
+			if err != nil {
+				return fmt.Errorf("failed to load okteto context '%s': %v", okteto.Context().Name, err)
+			}
+
+			return executeListPipelines(cmd.Context(), *flags, configmaps.List, getPipelineListOutput, c, os.Stdout)
+		},
+	}
+
+	cmd.Flags().StringArrayVarP(&flags.labels, "label", "", []string{}, "tag and organize dev environments using labels (multiple --label flags accepted)")
```
Comment (jLopezbarb) [inline cmd/pipeline/list.go:98]: Should we be able to run it into a certtain cluster?

```suggestion
	cmd.Flags().StringVarP(&flags.context, "context", "c", "", "context where the pipelines are deployed (defaults to the current context)")
	cmd.Flags().StringVarP(&flags.namespace, "namespace", "n", "", "namespace where the pipelines are deployed (defaults to the current namespace)")
```

File: cmd/pipeline/list.go [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+
+func list(ctx context.Context) *cobra.Command {
+	flags := &listFlags{}
+
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "List all okteto pipelines",
+		Args:  utils.NoArgsAccepted(""),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if flags.namespace == "" {
```
Comment (jLopezbarb) [inline cmd/pipeline/list.go:61]: Maybe we can wrap all the context initialisation in order to make it testable.

File: cmd/pipeline/list.go (hunk 1 above)
Comment (andreafalzetti) [reply cmd/pipeline/list.go:98]: Do we want to? I am not sure, considering other commands require switching context before executing them.

File: cmd/pipeline/list.go (hunk 2 above)
Comment (andreafalzetti) [reply cmd/pipeline/list.go]: I made some changes now, not perfect but still it's an improvement. Let me know if you have feedback on that

File: cmd/preview/list.go
Comment (teresaromero) [inline cmd/preview/list.go:1]: i did some changes here https://github.com/okteto/okteto/pull/3970 you might want to rebase, i just merged to master

File: cmd/pipeline/list.go (hunk 1 above)
Comment (teresaromero) [reply cmd/pipeline/list.go:65]: is the context flag supported by the spec? 🤔  @jLopezbarb

Comment (jLopezbarb) [reply cmd/pipeline/list.go:65]: > Do we want to? I am not sure, considering other commands require switching context before executing them.

I think it's the same for this one, we need to load a context given the namespace and the context (default to current one). We can talk about this offline before making a decision, but I think it's the same as the rest of pipeline commands

> is the context flag supported by the spec? 

No, but I find it odd to have that flag on all the commands except for this one.

File: cmd/preview/list.go
Comment (andreafalzetti) [reply cmd/preview/list.go:1]: I've tried rebasing from master but it's giving me a bunch of conflicts to fully rebase. Considering that PR is for `preview list` perhaps I can just go ahead and merge this one, or is there anything particular that I should make sure to check?

Comment (teresaromero) [reply cmd/preview/list.go:1]: i did some changes regarding testing and wrapping functions, the changes regarding make private the flag i did them already so maybe you can remove from ur files? 🤔  what kind of problem are there?

Comment (andreafalzetti) [reply cmd/preview/list.go:1]: @teresaromero ahh ok, I hadn't noticed! Fixed now

File: cmd/pipeline/list.go (hunk 1 above)
Comment (andreafalzetti) [reply cmd/pipeline/list.go:65]: as discussed offline, it makes sense to add, so I've updated the PR reflecting this change!!

**Review by teresaromero — APPROVED** [review]
👍🏻  Tested and looking good! Also tested the log-output combination to check it works as expected :) 👏🏻 👏🏻

**Pull request merged**